wheel/wheel.ino

```diff
@@ -39,208 +84,213 @@ GND	| GND
+
+  // Throttle Potentiometer
+  Pot throttle(34,0,4095,0,100, "Throttle"); // pin, analog min & max, out min & max; Scale => [0,100]
 
 void setup() {
-  Serial.begin(9600);
 
-  while(!Serial){}
+  Serial.begin(9600); // Can have Debug send to another stream if we want
+
+  Debug.setDebugLevel(debuggingLevel);
+  Debug.timestampOn();
+  Debug.newlineOn(); // Send a new line after every message
 
-  Serial.println("CAN mcu");
+  while(debuggingLevel >= DBG_DEBUG && !Serial){} // Wait for serial only if in debug mode or higher
+  DEBUG_INFO("CAN mcu");
 
   // Steering momentary buttons
-  pinMode(CommsPin, INPUT_PULLUP);
-  pinMode(GainUpPin, INPUT_PULLUP);
-  pinMode(TrimUpPin, INPUT_PULLUP);
-  pinMode(TrimDownPin, INPUT_PULLUP);
-  pinMode(GainDownPin, INPUT_PULLUP);
+  for (int i = 0; i < NUM_BUTTONS; i++) {
+    pinMode(buttons[i], INPUT_PULLUP);
+  }
 
-  // Encoder & Potentiometer should NOT be set to input
+  // Encoder & Potentiometer SHOULD NOT be set to input
   
   // Switchboard LEDs
-  pinMode(LED1, OUTPUT);
-  pinMode(LED2, OUTPUT);
-  pinMode(LED3, OUTPUT);
-  pinMode(LED4, OUTPUT);
-  pinMode(LED5, OUTPUT);
-  pinMode(LED6, OUTPUT);
-
-  if(!CAN.begin(500E3)){
-    Serial.println("Starting CAN fail");
-    while(1);
+  for (int i = 0; i < NUM_LEDS; i++) {
+    pinMode(LEDs[i], OUTPUT);
   }
 
-
+  // Wait for CAN to begin
+  while(!CAN.begin(bitrateCAN)){
+    DEBUG_WARNING("Starting CAN failure");
+  }
 }
 
 void loop() {
 
-  encoderVal = readEncoder();
-
-  // Encoder testing
-  Serial.print("Encoder: ");
-  Serial.println(encoderVal);
-
-  encodertoCAN(encoderVal);
-
-  throttleVal = readThrottle();
-
-  // Throttle testing
-  Serial.print("Throttle: ");
-  Serial.println(throttleVal);
-  delay(500);
+  encoderSend();
 
-  throttletoCAN(throttleVal);
+  throttleSend();
 
-  buttonstoCAN();
+  buttonsSend();
 
   LEDReceive();
 }
 
-// From https://esp32io.com/tutorials/esp32-potentiometer
-float floatMap(float x, float in_min, float in_max, float out_min, float out_max) {
-  return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
-}
 
-float readEncoder(){
-    return floatMap(analogRead(ThrottlePin), 0, 4095, 0, 360); //Scale => [0,360]
+// Encoder and Throttle to CAN
+void potToCAN(int id, int val, const String name) {
+  size_t size = sizeof(val);
+  CAN.beginPacket(id);
+  uint8_t data[size];                   // Create uint8_t array
+  memcpy(data, &val, size);             // Store bytes of val to array
+  size_t sent = CAN.write(data, size);  // Write the buffer to CAN
+  int end = CAN.endPacket();
+
+  DEBUG_INFO("%s - %d", name, val);
+  if (sent < size) {
+    DEBUG_WARNING("Size of message sent < size of data");
+  } else {
+    DEBUG_INFO("All data sent");
+  }
+  DEBUG_VERBOSE("endPacket output: %d", end);
 
 }
 
-float readThrottle(){
-  return floatMap(analogRead(ThrottlePin), 0, 4095, 0, 100); //Scale => [0,100]
+void encoderSend() {

```

Comment:
why have a function that just calls another func

Reply:
So that I can pass different values depending on the potentiometer in the future, but don't need to.